engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/flink/test/FlinkMiniClusterHolder.scala

```diff
@@ -2,17 +2,20 @@ package pl.touk.nussknacker.engine.flink.test
 
 import java.util.concurrent.CompletableFuture
 
-import org.apache.flink.api.common.JobID
+import org.apache.flink.api.common.{JobID, JobStatus}
 import org.apache.flink.client.program.ClusterClient
-import org.apache.flink.configuration.{ConfigConstants, ConfigOptions, Configuration, CoreOptions, QueryableStateOptions, TaskManagerOptions}
+import org.apache.flink.configuration._
 import org.apache.flink.queryablestate.client.QueryableStateClient
+import org.apache.flink.runtime.client.JobStatusMessage
 import org.apache.flink.runtime.executiongraph.AccessExecutionGraph
+import org.apache.flink.runtime.jobgraph.JobGraph
 import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration
 import org.apache.flink.test.util.MiniClusterWithClientResource
-import org.scalatest.concurrent.Eventually.scaled
+import org.scalatest.concurrent.Eventually.{scaled, _}
 import org.scalatest.time.{Millis, Seconds, Span}
-import FlinkMiniClusterHolder._
-import org.scalatest.concurrent.Eventually._
+import pl.touk.nussknacker.engine.flink.test.FlinkMiniClusterHolder._
+
+import scala.collection.JavaConverters._
 
 // This class is splitted into trait and Impl because of Flink's API changes between 1.6 and 1.7 version
 trait FlinkMiniClusterHolder {
```

Comment:
Can you add a comment that it is class that should be used once for more then one job because staring/stopping of cluster is costly. And the second purpose for this class is to handle compatibility issues between different versions of Flink...

Reply:
Done